maintenance_team_hierarchy/models/maintenance_team.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MaintenanceTeam(models.Model):
+
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = "name"
+    _inherit = "maintenance.team"
+
+    parent_id = fields.Many2one("maintenance.team", ondelete="restrict")
+    parent_left = fields.Integer(index=True)
+    parent_right = fields.Integer(index=True)
+    request_ids = fields.Many2many(
+        "maintenance.request", compute="_compute_request_ids"
+    )
+
+    def _compute_request_ids(self):
+        for record in self:
+            record.request_ids = self.env["maintenance.request"].search(
+                record._get_request_domains()
+            )
+
+    def _get_request_domains(self):
+        return [("maintenance_team_id", "child_of", self.id)]
+
+    @api.one
+    @api.depends()
+    def _compute_todo_requests(self):
+        """This function improves the computation and simplifies the data"""
```

Comment:
This function is necessary as the current behaviour was extreamly slowly... For example, if we have 20.000 requests on a team, it took 2 seconds to do the query,  now it takes less than 0.1 seconds.

Reply:
Nice, but two comments. Why did you remove the `@api.depends('request_ids.stage_id.done')`? And I think we should put a context to call the old function just in case, the same we always do when overriding a function.